src/settings.h

```diff
@@ -44,12 +45,28 @@ class CSettings : public QObject
     Q_OBJECT
 
 public:
-    CSettings() :
```

Comment:
This seems a bit odd to me. Are you sure this isn't just fixing symptoms?
Concept wise, the settings and gui shouldn't be coupled that closely.

Reply:
I guess it could be done with something like this in `main.cpp`:
```
// ... get the app object to pApp or whatever
// ... get the settings object to pSettings or whatever
#ifndef HEADLESS
        if ( bUseGUI )
        {
            const QGuiApplication* pGApp = static_cast<const QGuiApplication*> ( pApp );

            QObject::connect ( pGApp, &QGuiApplication::saveStateRequest, this, [=] ( QSessionManager& ) { pSettings->Save(); );

            QObject::connect ( pGApp, &QGuiApplication::applicationStateChanged, this, [=] ( Qt::ApplicationState state ) {
                if ( Qt::ApplicationActive != state )
                {
                    pSettings->Save();
                }
            } );
        }
#endif
```

Reply:
OK, tested the new approach and it worked as expected.